From d3e295de20166f1c91125309756bf24f2211d1df Mon Sep 17 00:00:00 2001
From: HuyLe <hule@amperecomputing.com>
Date: Mon, 18 Mar 2024 08:58:01 +0700
Subject: [PATCH 1/3] Add show info about password Ampere policy

Update WebUI help text for password policy:
- Password length between 9 and 20 characters
- Must have at least 1 lowercase, 1 uppercase, 1 number and 1 special character
- No more than 3 consecutive characters
- Other patterns that libpam rejects

Change-Id: I94e0d6f7c3aa8f185279442377143c2d09734916
Signed-off-by: HuyLe <hule@amperecomputing.com>
---
 src/locales/en-US.json                                   | 3 +++
 src/views/SecurityAndAccess/UserManagement/ModalUser.vue | 6 ++++++
 2 files changed, 9 insertions(+)

diff --git a/src/locales/en-US.json b/src/locales/en-US.json
index 8ba7ac9..6d0cf05 100644
--- a/src/locales/en-US.json
+++ b/src/locales/en-US.json
@@ -569,6 +569,9 @@
             "maxFailedLoginAttempts": "Max failed login attempts",
             "noSpecialCharactersExceptUnderscore": "No special characters except underscore",
             "passwordMustBeBetween": "Password must be between %{min} – %{max} characters",
+            "mustHaveAtLeast": "No more than 3 consecutive characters",
+            "noMoreThan": "Contain at least 1 lowercase, 1 upper case, 1 number and 1 special character",
+            "openBMCPolicies": "Other default OpenBMC password policy (not listed here)",
             "passwordsDoNotMatch": "Passwords do not match",
             "privilege": "Privilege",
             "timeoutDurationSeconds": "Timeout duration (seconds)",
diff --git a/src/views/SecurityAndAccess/UserManagement/ModalUser.vue b/src/views/SecurityAndAccess/UserManagement/ModalUser.vue
index 5ffb297..15c746d 100644
--- a/src/views/SecurityAndAccess/UserManagement/ModalUser.vue
+++ b/src/views/SecurityAndAccess/UserManagement/ModalUser.vue
@@ -137,6 +137,12 @@
                     max: passwordRequirements.maxLength,
                   })
                 }}
+                <br />
+                {{ $t('pageUserManagement.modal.mustHaveAtLeast') }}
+                <br />
+                {{ $t('pageUserManagement.modal.noMoreThan') }}
+                <br />
+                {{ $t('pageUserManagement.modal.openBMCPolicies') }}
               </b-form-text>
               <input-password-toggle>
                 <b-form-input
-- 
2.34.1

